src/sagemaker/workflow/utilities.py

```diff
@@ -13,30 +13,37 @@
 """Utilities to support workflow."""
 from __future__ import absolute_import
 
+import logging
+from functools import wraps
 from pathlib import Path
-from typing import List, Sequence, Union, Set
+from typing import List, Sequence, Union, Set, TYPE_CHECKING
 import hashlib
 from urllib.parse import unquote, urlparse
 from _hashlib import HASH as Hash
 
+from sagemaker.workflow.parameters import Parameter
 from sagemaker.workflow.pipeline_context import _StepArguments
-from sagemaker.workflow.step_collections import StepCollection
 from sagemaker.workflow.entities import (
     Entity,
     RequestType,
 )
 
+if TYPE_CHECKING:
```

Comment:
interesting, so `type_checking` resolves the circular import error from importing `StepCollection` here?

Reply:
Yes